### PR TITLE
Scoreboard

### LINKED ISSUE #17: Slash Command - Score Tracking
`/track <score-name>` Begins tracking a score for the user that runs the command
`/update <score-name> <amount>` Adds `amount` to `score-name` for the issuer (can be negative or fractional)
`/list-tracking` list all of the `score-name`s being tracked for the issuer
`/stop-tracking <score-name>` erase score for the issuer on `score-name`
`/leaderboard <score-name>` show the scores of all users for `score-name` that are being tracked for `score-name`

### PR DESCRIPTION
**Pull request by Plyb**
Adds the `/stats` command and resolves #17

### CODE REVIEW

File: src/commands/stats.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+): Promise<void> {
+	const statName = sanitize(interaction.options.getString(StatNameOption));
+	const amount = interaction.options.getNumber(AmountOption);
+
+	if (statName === undefined) {
+		throw mustIncludeNameError();
+	}
+	if (amount === null) {
+		throw new UserMessageError('Must include amount');
+	}
```
Comment (AverageHelper) [inline src/commands/stats.ts:157]: Same here:

```suggestion
	const statName = sanitize(interaction.options.getString(StatNameOption, true));
	const amount = interaction.options.getNumber(AmountOption, true);
```

File: src/commands/stats.ts [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+async function untrack(
+	reply: InteractionContext['reply'],
+	interaction: ChatInputCommandInteraction,
+	guildId: string
+): Promise<void> {
+	const statName = sanitize(interaction.options.getString(StatNameOption));
+
+	if (statName === undefined) {
+		throw mustIncludeNameError();
+	}
```
Comment (AverageHelper) [inline src/commands/stats.ts:226]: Same here:

```suggestion
	const statName = sanitize(interaction.options.getString(StatNameOption, true));
```

File: src/commands/stats.ts [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+	reply: InteractionContext['reply'],
+	interaction: ChatInputCommandInteraction,
+	guildId: string,
+	userCache: Collection<string, User>
+): Promise<void> {
+	const statName = sanitize(interaction.options.getString(StatNameOption));
+
+	if (statName === undefined) {
+		throw mustIncludeNameError();
+	}
```
Comment (AverageHelper) [inline src/commands/stats.ts:262]: Same here:

```suggestion
	const statName = sanitize(interaction.options.getString(StatNameOption, true));
```

File: src/commands/stats.ts [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+	});
+}
+
+function notCurrentlyTrackingError(statName: string): UserMessageError {
+	return new UserMessageError(
+		`I'm not currently tracking "${statName}" for you. Use /track to begin tracking that score.`
+	);
+}
+
+function mustIncludeNameError(): UserMessageError {
```
Comment (AverageHelper) [inline src/commands/stats.ts:311]: With the above suggestions, this function is unused, and may be removed.

**Review by AverageHelper — CHANGES_REQUESTED** [review]

**Review by ZYancey — APPROVED** [review]

**Review by AverageHelper — APPROVED** [review]
LGTM! What's the plan for migrating from Ze Kaiser's scoreboard?

**Comment by Plyb** [comment]
I've got a copy of Ze Kaiser's scoreboard in JSON format from Arken. I'll use a script to port it over

**Pull request merged**